nibabel/spatialimages.py

```diff
@@ -152,21 +185,33 @@ class HeaderTypeError(Exception):
     """Class to indicate error in parameters into header functions"""
 
 
-class SpatialHeader(FileBasedHeader):
+class SpatialHeader(FileBasedHeader, SpatialProtocol):
     """Template class to implement header protocol"""
 
-    default_x_flip = True
-    data_layout = 'F'
+    default_x_flip: bool = True
+    data_layout: Literal['F', 'C'] = 'F'
+
+    _dtype: np.dtype
+    _shape: tuple[int, ...]
+    _zooms: tuple[float, ...]
 
-    def __init__(self, data_dtype=np.float32, shape=(0,), zooms=None):
+    def __init__(
+        self,
+        data_dtype: npt.DTypeLike = np.float32,
+        shape: Sequence[int] = (0,),
+        zooms: Sequence[float] | None = None,
```

Comment:
These should be replaced with `collections.abc.Sequence` at some point, when support for 3.8 is dropped. I realize there's time, but perhaps we can keep track of it somewhere? Or should we just wait for the errors that will be raised in some future Python version?

Reply:
Made a suggestion above.